akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -293,6 +293,51 @@ private[akka] object StreamLayout {
   }
 }
 
+private[stream] object V {
+  sealed trait Termination
+  case object Allowed extends Termination
+  case object Completed extends Termination
+  case class Failed(ex: Throwable) extends Termination
+}
+
+private[stream] final class V[T] extends Processor[T, T] {
+  import V._
+  import ReactiveStreamsCompliance._
+
+  private val subscriptionStatus = new AtomicReference[AnyRef]
+  private val terminationStatus = new AtomicReference[Termination]
+
+  override def subscribe(s: Subscriber[_ >: T]): Unit =
+    if (subscriptionStatus.compareAndSet(null, s)) () // wait for onSubscribe
+    else {
+      val sub = subscriptionStatus.getAndSet(s).asInstanceOf[Subscription]
+      tryOnSubscribe(s, sub)
+      terminationStatus.getAndSet(Allowed) match {
```

Comment:
this will be null if onSubscribe was first, since it does not set terminationStatus after the successful CAS on subscriptionStatus, and this will enter this branch since the CAS will fail here on subscriptionStatus. Maybe you wanted this to have the initial value of Allowed?